File: refactor_imports.py

import os

project_root = r"c:\Users\frank\Downloads\Zync-mobile"
replacements = {
    "@/presentation/components/ui/": "@/components/",
    "@/presentation/components/": "@/components/",
    "@/application/ZyncContext": "",
    "@/application/CartContext": "@/features/wallet/context/CartContext",
    "@/infrastructure/spotify-service": "@/features/music/services/spotify-service",
    "@/presentation/components/TicketCard": "@/features/dashboard/components/TicketCard",
    "@/presentation/components/ui/PaymentCard": "@/features/wallet/components/PaymentCard",
}

for root, dirs, files in os.walk(project_root):
    # skip node_modules and .git
    if "node_modules" in root or ".git" in root:
        continue
        
    for file in files:
        if file.endswith(".tsx") or file.endswith(".ts"):
            filepath = os.path.join(root, file)
            try:
                with open(filepath, "r", encoding="utf-8") as f:
                    content = f.read()
                
                new_content = content
                for old, new in replacements.items():
                    new_content = new_content.replace(old, new)
                
                if new_content != content:
                    with open(filepath, "w", encoding="utf-8") as f:
                        f.write(new_content)
                    print(f"Updated {filepath}")
            except Exception as e:
                print(f"Error processing {filepath}: {e}")

File: refactor_imports.js

const fs = require('fs');
const path = require('path');

const projectRoot = 'c:\\Users\\frank\\Downloads\\Zync-mobile';
const replacements = {
    '@/presentation/components/ui/': '@/components/',
    '@/presentation/components/': '@/components/',
    '@/application/ZyncContext': '@/context/ZyncContext',
    '@/application/CartContext': '@/features/wallet/context/CartContext',
    '@/infrastructure/spotify-service': '@/features/music/services/spotify-service',
    '@/presentation/components/TicketCard': '@/features/dashboard/components/TicketCard',
    '@/presentation/components/ui/PaymentCard': '@/features/wallet/components/PaymentCard',
    // Fix relative paths if any (simple heuristic)
    '../../presentation/components/': '@/components/',
    '../presentation/components/': '@/components/'
};

function walk(dir) {
    const list = fs.readdirSync(dir);
    list.forEach(file => {
        const filePath = path.join(dir, file);
        const stat = fs.statSync(filePath);
        if (stat.isDirectory()) {
            if (file !== 'node_modules' && file !== '.git') walk(filePath);
        } else {
            if (file.endsWith('.tsx') || file.endsWith('.ts')) {
                let content = fs.readFileSync(filePath, 'utf8');
                let original = content;
                for (const [oldVal, newVal] of Object.entries(replacements)) {
                    content = content.split(oldVal).join(newVal);
                }
                if (content !== original) {
                    fs.writeFileSync(filePath, content, 'utf8');
                    console.log(`Updated ${filePath}`);
                }
            }
        }
    });
}

walk(projectRoot);
